### PR TITLE
[telemetry] Added build flavours for OkHttp v3.x and v4.x

### PR DESCRIPTION
**Pull request by tarigo**
Added 2 new build flavours producing the following arctifacts:
1. `com.mapbox.mapboxsdk:mapbox-android-telemetry:${version}` - by default, old artifact name, comes with OkHttp v4.x and minSdkVersion == 21. 
2. `com.mapbox.mapboxsdk:mapbox-android-telemetry-okhttp3:${version}` - old configuration: OkHttp v3.x and minSdkVersion == 14.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -69,7 +69,7 @@ jobs:
           no_output_timeout: 20m
           command: |
             build_dir="libtelemetry/build"
-            test_apk_path="outputs/apk/androidTest/debug/libtelemetry-debug-androidTest.apk"
+            test_apk_path="outputs/apk/androidTest/okhttp4/debug/libtelemetry-okhttp4-debug-androidTest.apk"
```
Comment (mr1sunshine) [inline .circleci/config.yml:72]: Don't we want to keep the same behavior here? I mean having okhttp4 versioned telemetry without any suffixes? Even example apps.

**Review by mr1sunshine — APPROVED** [review]
LGTM

File: .circleci/config.yml (hunk 1 above)
Comment (tarigo) [reply .circleci/config.yml:72]: Unfortunately, it's generated by Google's Gradle android build tools according to build flavour names and I don't know how to override build outputs.

File: cloud_test.sh [hunk 2]
```diff
@@ -7,7 +7,7 @@ test_apk_path="$2"
 results_dir="$3"
 gcloud firebase test android models list
 gcloud firebase test android run --type instrumentation \
-    --app app/build/outputs/apk/debug/app-debug.apk \
+    --app app/build/outputs/apk/okhttp4/debug/app-okhttp4-debug.apk \
```
[the remark this answers is not in the document]
Comment (tarigo) [reply cloud_test.sh:10]: ^ in the answer above.

**Pull request merged**